master/security/icmp-ping.md

```diff
@@ -0,0 +1,136 @@
+
+For details, see [ICMP type and code](https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol#Control_messages).
+
+### How to
+
+- [Deny all ICMP, all workloads](#deny-all-icmp, all workloads)
+- [Allow ICMP ping, all workloads and host endpoints](#allow-icmp-ping-all-workloads-and-host-endpoints)
+- [Allow ICMP matching protocol type and code, all workloads](#allow-icmp-matching-protocol-type-and-code-all-workloads)
+
+#### Deny all ICMP, all workloads
+
+In this example, we introduce a "deny all ICMP" **GlobalNetworkPolicy**. 
+
+This policy **selects all workloads that match a selector**. It enables a default deny for all workloads, in addition to the explicit ICMP deny rules specified in the policy. For more on default deny policy, see [Calico Network Policy]({{site.baseurl}}/{{page.version}}/security/caliconetworkpolicy).   
+
+If your ultimate goal is to allow some traffic, have your regular "allow" policies in place before applying a global deny-all ICMP traffic policy.
+
+In this example, all workloads are blocked from sending or receiving **ICMPv4** and **ICMPv6** messages. Because a GlobalNetworkPolicy applies to hostendpoints as well, to limit the policy to only workloads, we use the selector **projectcalico.org/orchestrator == 'kubernetes'**.
+
+If **ICMPv6** messages are not used in your deployment, it is still good practice to deny them specifically as shown below. 
+
+In any "deny-all" Calico network policy, be sure to specify a lower order (**order:200**) than regular policies that might allow traffic.   
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: block-icmp
+spec:
+  order: 200
+  selector: projectcalico.org/orchestrator == 'kubernetes'
+  types:
+  - Ingress
+  - Egress
+  ingress:
+  - action: Deny
+    protocol: ICMP
+  - action: Deny
+    protocol: ICMPv6
+  egress:
+  - action: Deny
+    protocol: ICMP
+  - action: Deny
+    protocol: ICMPv6    
+```
+
+#### Allow ICMP ping, all workloads and host endpoints
+
+In this example, workloads can receive **ICMPv4 type 8** and **ICMPv6 type 128** ping requests.
+
+All other traffic may be allowed by other policies. If traffic is not explicitly allowed, it will be denied by default. 
+
+The policy applies only to **ingress** traffic. (Egress traffic is not affected, and default deny is not enforced for egress.) 
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: allow-ping-in-cluster
+spec:
+  selector: all()
+  types:
+  - Ingress
+  ingress:
+  - action: Allow
+    protocol: ICMP
+    source:
+      selector: all()
+    icmp:
+      type: 8 # Ping request
+  - action: Allow
+    protocol: ICMPv6
+    source:
+      selector: all()
+    icmp:
+      type: 128 # Ping request
+```
+
+#### Allow ICMP matching protocol type and code, all workloads
```

Comment:
> all workloads

All workloads is subtly different from all pods. In this case, we match all Kubernetes pods but not all workloads (for example, this wouldn't select open stack VMs).

Reply:
Fixed.